contracts/floaty/src/instructions.rs

```diff
@@ -0,0 +1,551 @@
+use cosmwasm_schema::cw_serde;
+use rand_chacha::rand_core::RngCore;
+
+use crate::floats::{random_f32, random_f64};
+
+/// Not intended for direct usage
+macro_rules! run_instr {
+    ($instr:expr, $input:expr, $input_ty:ty, $return_ty:ty) => {{
+        let input: $input_ty = $input;
+        let ret: $return_ty;
+        unsafe {
+            core::arch::asm!("local.get {0}", $instr, "local.set {1}", in(local) input, out(local) ret)
+        };
+        ret
+    }};
+    ($instr:expr, $input1:expr, $input1_ty:ty, $input2:expr, $input2_ty:ty, $returnty:ty) => {{
```

Comment:
(this is nitting, sorry for that)
I would change `$returnty` to `$return_ty` to be defined the same way as in previous matching clause.